src/app/profile/directives/box-color.directive.ts

```diff
@@ -0,0 +1,24 @@
+import { Directive, ElementRef, Input, OnChanges } from '@angular/core';
+
+@Directive({
+  selector: '[appBoxColor]',
+})
+export class BoxColorDirective implements OnChanges {
+  @Input() appBoxColor = '';
+
+  constructor(private el: ElementRef) {}
+
+  ngOnChanges(): void {
+    const creationDate = new Date(this.appBoxColor).getTime();
+    const currentDate = new Date();
+    const currentDateTime = currentDate.getTime();
+
+    if (creationDate < currentDateTime) {
```

Comment:
### Issue
1) It would be more appropriate to create a variable for storing the desired colour and changing it based on your condition.
2) What happens if the `creationDate` and `curretnDateTime` are equal?

### Suggested change
```typescript
ngOnChanges(): void {
  const creationDate = new Date(this.appBoxColor).getTime();
  const currentDateTime = new Date().getTime();

  this.el.nativeElement.style.boxShadow =
    creationDate < currentDateTime
      ? `5px 5px 10px #45455b`
      : `5px 5px 10px #a4bc41`;
}

Reply:
reworked, but without ternary operator